lib/memsize_helpers.rb

```diff
@@ -1,6 +1,40 @@
+require "objspace"
+
 module MemsizeHelpers
   def self.memsize_rss_in_kb
-    _, size = %x(ps -p "#{$$}" -o pid=,rss=).strip.split.map(&:to_i)
-    size
+    %x(ps -p "#{$$}" -o rss=).strip.to_i
+  end
+
+  def self.reset!
+    @memsize_rss_start = memsize_rss_in_kb
+  end
+  reset!
+
+  def self.check_usage
+    GC.start
+    if verbose?
+      # Put newline first to keep these aligned in the middle of the benchmkark-ips output.
+      printf "\n[Memory usage: %11s KB - ruby space %7s KB]",
+        format_number(MemsizeHelpers.memsize_rss_in_kb),
+        format_number((ObjectSpace.memsize_of_all / 1000).round(0))
+    end
+  end
+
+  def self.format_number(n)
```

Comment:
Is there a more detailed name we can use? I'm finding this much harder to read than the previous code that just used string interpolation. I don't doubt that this is better, but it's harder to follow.

Rails has a nice helper for formatting sizes. Maybe we can use its name. But, I appreciate not pulling in that large dependency. I'd like to keep this as slim as possible.

Reply:
Moved to https://github.com/nirvdrum/grpc-protobuf-experiments/pull/3 and renamed to "format_with_delimiter".
Rails has a number to delimited converter.